flowermd/base/simulation.py

```diff
@@ -1085,6 +1128,74 @@ def save_restart_gsd(self, file_path="restart.gsd"):
         """
         hoomd.write.GSD.write(self.state, filename=file_path)
 
+    def save_simulation(self, file_path="simulation.pickle", save_walls=False):
+        """Save a pickle file with everything needed to retart a simulation.
+
+        This method is useful for saving the state of a simulation to a file
+        and reusing it for restarting a simulation or running a different
+        simulation.
+
+        Parameters
+        ----------
+        file_path : str, default "simulation.pickle"
+            The path to save the pickle file to.
+        save_walls : bool, default False
+            Determines if any wall forces are saved.
+
+        Notes
+        -----
+        This method creates a dictionary that contains the
+        simulation's forcefield, references values, and snapshot.
+        The key:value pairs are:
+
+            'reference_values': dict of str:float
+            'forcefield': list of hoomd forces
+            'state': gsd.hoomd.Frame
+            'sim_kwargs': dict of flowermd.base.Simulation kwargs
+
+        Wall forces are not able to be reused when starting
```

Comment:
I'm wondering if the wall forces can't be reused, why we need to pickle them in the first place?

Reply:
I agree, but I feel like we should at least include the option, that way the wall information still recorded and saved and accessible if you're restarting a simulation later.  Maybe we should check and throw an error on warning in `from_simulation_pickle`.  I also haven't done a dep dive into what the actual issue is, I just know hoomd throws an error.

Reply:
Ok, looking at this now. If we create a wall force in the simulation, the wall origin is a HOOMD tuple, but if we pickle and reload a wall force, then the wall origin is a regular tuple.

In the simulation where the wall force was first created

`_HOOMDTuple(21.28712272644043, 0.0, 0.0)`

In the simulation that loaded forces from a pickle

`(21.28712272644043, 0.0, 0.0)`

Reply:
I see. Good catch. 
Hoomd has an internal function called `[to_hoomd_data](https://github.com/glotzerlab/hoomd-blue/blob/82938e56fc9dfd703a3e5cf68c5391915841ab70/hoomd/data/collections.py#L275)` that I think can convert a regular tuple to HoomdTuple.

Reply:
Now, when saving a simulation, walls are always saved as well, and when loading from the resulting pickle file, the wall objects are recreated. This seems to work fine. I left the save_walls option for pickling the forcefield though.